src/module.ts

```diff
@@ -29,13 +31,24 @@ export default defineNuxtModule<ModuleOptions>({
   },
   async setup(options, nuxt) {
     const resolver = createResolver(import.meta.url)
+    // opt-out early
+    if (!options.enabled) {
+      return
+    }
 
     // add options to runtime config
-    nuxt.options.runtimeConfig.public.nuxtRateLimit = defu(
-      nuxt.options.runtimeConfig.public.nuxtRateLimit,
-      options
+    nuxt.options.runtimeConfig.nuxtRateLimit = defu(
+      nuxt.options.runtimeConfig.nuxtRateLimit,
+      { headers: options.headers }
     )
 
+    // merge with route rules, so we get free route matching with baseURL support
+    for (const [route, rules] of Object.entries(options.routes)) {
+      nuxt.options.nitro.routeRules = defu(nuxt.options.nitro.routeRules, {
+        [route]: { ['nuxt-rate-limit']: { ...rules, route } },
```

Comment:
i'd normally use a more generic key but if Nitro implements rate limiting it will cause issues